segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -266,16 +271,21 @@ public QueueStats getQueueStatistics() {
      * Write Processor main loop. This method is not thread safe and should only be invoked as part of the Write Processor.
      */
     private void processWritesSync() {
-        if (getWriteLedger().ledger.isClosed()) {
-            // Current ledger is closed. Execute the rollover processor to safely create a new ledger. This will reinvoke
-            // the write processor upon finish, so
-            this.rolloverProcessor.runAsync();
-        } else if (!processPendingWrites()) {
-            // We were not able to complete execution of all writes. Try again.
-            this.writeProcessor.runAsync();
-        } else {
-            // After every run, check if we need to trigger a rollover.
-            this.rolloverProcessor.runAsync();
+        try {
+            if (getWriteLedger().ledger.isClosed()) {
+                // Current ledger is closed. Execute the rollover processor to safely create a new ledger. This will reinvoke
+                // the write processor upon finish, so
+                this.rolloverProcessor.runAsync();
+            } else if (!processPendingWrites()) {
+                // We were not able to complete execution of all writes. Try again.
+                this.writeProcessor.runAsync();
+            } else {
+                // After every run, check if we need to trigger a rollover.
+                this.rolloverProcessor.runAsync();
+            }
+        } catch (Exception ex) {
+            log.error("{}: processWritesSync failed.", this.traceObjectId, ex);
```

Comment:
Because this method is executed directly by an `ExecutorService`. Nobody else in my code is listening. I wanted to know if we have problems processing writes, but at the same time I didn't want to break the invariant (that was there before) which let the exceptions bubble out.